src/boundaries/stagger_and_step.jl

```diff
@@ -0,0 +1,255 @@
+                search radius δ₀.")
+    end
+
+    v = Vector{typeof(current_state(ds))}(undef, N)
+    v[1] = xi
+    for n in 1:N
+        show_progress && ProgressMeter.update!(progress, n)
+        if escape_time!(ds, xi, isinside; max_escape_time) > Tm
+            reinit!(ds, xi)
+        else
+            xp, Tp = stagger!(ds, xi, δ, Tm, isinside; stagger_mode, max_steps, γ, max_escape_time, rng)
+            # The stagger step may fail. We reinitiate the algorithm from a new initial condition.
+            if Tp < 0
+                xp = stagger_trajectory(ds, x0, Tm, isinside; δ₀, stagger_mode = :exp,max_steps, γ, max_escape_time, rng) 
+                if isnothing(xp)
+                    error("Cannot find a stagger trajectory. Choose a different starting 
+                          point or search radius δ₀.")
+                end
+                δ = 0.1
+            end
+            reinit!(ds, xp)
+        end 
+        step!(ds)
+        xi = copy(current_state(ds))
+        v[n] = xi
+    end
+    return StateSpaceSet(v)
+end
+
+
+"""
+    stagger_trajectory(ds, x0, Tm, isinside; kwargs...) -> xi
+
+On success, the function returns a point `xi` with the property `T(xi) > 
+Tm` with a random walk search around the initial coordinates 
+`x0`. `T(xi)` is the escape time of the initial condition `xi` from 
+the bounding box defined by the function `isinside`.  
+In the case where the algorithm cannot find a suitable point, the algorithm 
+returns nothing. 
+
+This is an auxiliary function for [`stagger_and_step`](@ref). 
+Keyword arguments and definitions are identical for both functions. 
+"""
+function stagger_trajectory(ds, x0, Tm, isinside; δ₀ = 1., stagger_mode = :exp,
+        max_steps = Int(1e5), γ = 1.1, max_escape_time = 10000, rng::AbstractRNG = Xoshiro())
+    T = escape_time!(ds, x0, isinside; max_escape_time)
+    xi = copy(x0) 
+    while !(T > Tm)  # we must have T > Tm at each step 
+        xi, T = stagger!(ds, xi, δ₀, T, isinside; γ, stagger_mode, max_steps, max_escape_time, rng)
+        if T < 0
+            return nothing
+        end 
+            
+    end
+    return xi
+end
+    
+
+
+function escape_time!(ds, x0, isinside; max_escape_time = 10000) 
+    set_state!(ds,x0)
+    reinit!(ds, x0)
+    k = 1; 
+    while isinside(current_state(ds)) 
+        if k > max_escape_time 
+            error("The trajectory did not escape for ", k, " steps, you probably have \
+an attractor in the defined region. Last point evaluated: ", current_state(ds))
+        end
+        step!(ds)
+        k += 1
+    end
+    return current_time(ds)
+end
+
+function rand_u!(u, δ, n, stagger_mode, rng)
+    if stagger_mode == :exp 
+        a = -log10(δ)
+        s = (15-a)*rand(rng) + a
+        u .= randn(rng,n)
+        u .*= (10.0^(-s))/norm(u)
+        return
+    elseif stagger_mode == :unif
+        s = δ*rand(rng)
+        u .= randn(rng,n)
```

Comment:
```suggestion
        randn!(rng, u)
```

this still allocates!!! You have to use in-place operations. Please read the documentation strings of `randn` and `randn!`. You need to do `using Random: randn!`.